web/pages/leaderboards.tsx

```diff
@@ -30,31 +40,69 @@ export default function Leaderboards(props: {
     topTraders: [],
     topCreators: [],
   }
-  const { topTraders, topCreators } = props
+
+  const [topTradersState, setTopTraders] = useState(props.topTraders)
+  const [topCreatorsState, setTopCreators] = useState(props.topCreators)
+  const [isLoading, setLoading] = useState(false)
+  const [period, setPeriod] = useState<LeaderboardPeriod>('weekly')
+
+  useEffect(() => {
+    setLoading(true)
+    queryTopCreators(period).then((res) => {
+      setTopTraders(res.props.topTraders as User[])
+      setTopCreators(res.props.topCreators as User[])
+      setLoading(false)
+    })
+  }, [period])
 
   return (
     <Page margin>
+      <Title text={'Leaderboards'} className={'hidden md:block'} />
+      <Col className="gap-4 text-center">
+        <select
```

Comment:
Very nice. Consider using Tabs instead of a select (see tabs.tsx). But up to you, I haven't checked how it looks yet!